Lib/zipfile/__init__.py

```diff
@@ -1781,11 +1781,14 @@ def _extract_member(self, member, targetpath, pwd):
         # Create all upper directories if necessary.
         upperdirs = os.path.dirname(targetpath)
         if upperdirs and not os.path.exists(upperdirs):
-            os.makedirs(upperdirs)
+            os.makedirs(upperdirs, exist_ok=True)
 
         if member.is_dir():
             if not os.path.isdir(targetpath):
-                os.mkdir(targetpath)
+                try:
+                    os.mkdir(targetpath)
+                except FileExistsError:
+                    pass
```

Comment:
Currently it raises FileExistsError if targetpath exists, but is not a directory. With this change it will be silent. It is not good, because it is an error.

Please add a test for extracting a directory in place of existing regular file.

Reply:
Such test was added in #114960, and it fails with this PR.